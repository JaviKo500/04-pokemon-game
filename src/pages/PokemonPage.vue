<template>
    <h1 v-if="!pokemon" >await please ...... </h1>
    <div class="" v-else>
        <h1>what is this pokemon?</h1>
        <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon" />
        <PokemonOptions :pokemons="pokemonsArr"/>
    </div>
</template>

<script>
import PokemonPicture from '@/components/PokemonPicture'
import PokemonOptions from '@/components/PokemonOptions'
import getPokemonsOptions from '@/helpers/getPokemonOptions'
export default {
    components: {
        PokemonPicture,
        PokemonOptions,
    },
    data() {
        return {
            pokemonsArr: [],
            pokemon: null,
            showPokemon: false
        }
    },
    mounted(){
        this.mixPokemonsArr();
    },
    methods: {
        async mixPokemonsArr(){
            this.pokemonsArr = await getPokemonsOptions();
            const rnsInt = Math.floor( Math.random() * 4 );
            this.pokemon = this.pokemonsArr[rnsInt];
        }
    },
    
}
</script>
<!-- test data -->
<style>

</style>