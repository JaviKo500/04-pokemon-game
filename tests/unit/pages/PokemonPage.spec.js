import { shallowMount, mount } from "@vue/test-utils";
import PokemonPage from '@/pages/PokemonPage'
import { pokemons } from "../mocks/pokemons.mock";

describe('PokemonPage component', () => {
    let wrapper
    let mixPokemonsArr;
    beforeEach(()=>{
        // mixPokemonsArr = jest.spyOn(PokemonPage.methods, 'mixPokemonsArr');
        wrapper = shallowMount( PokemonPage );
    });

    test( 'do match with snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot(); 
    }); 

    test( 'should call mixPokemonsArr to mount', () => {
        const mixPokemonsArr = jest.spyOn(PokemonPage.methods, 'mixPokemonsArr');
        shallowMount(PokemonPage);
        expect( mixPokemonsArr ).toHaveBeenCalled( );
    });

    test( 'do match snapshot load pokemons', () => {
        // const wrapper = mount( PokemonPage, {
        const wrapper = shallowMount( PokemonPage, {
            data() {
                return {
                    pokemonsArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        } );

        expect( wrapper.html() ).toMatchSnapshot();

    });

});