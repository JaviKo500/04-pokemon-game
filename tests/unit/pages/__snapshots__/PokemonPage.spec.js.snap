// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage component do match snapshot load pokemons 1`] = `
<div class="">
  <h1>what is this pokemon?</h1>
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
  <!--v-if-->
</div>
`;

exports[`PokemonPage component do match with snapshot 1`] = `<h1>await please ...... </h1>`;
